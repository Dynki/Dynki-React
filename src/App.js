import React, { Component } from 'react';
import { connect } from 'react-redux'
import { setDomain } from './store/actions/authActions'

import './App.scss';
import { SecuredRoute, PostAuthShell } from './components';
import MainErrorBoundary from './components/core/MainErrorBoundry';

export class App extends Component {

  constructor(props) {
    super(props)

    this.state = {
      domainLoaded: false
    }
  }

  componentDidMount() {
    this.onSetDomain()
  }

  onSetDomain() {
    this.props.SetDomain();
    this.setState({ domainLoaded: true })
  }

  render() {
    const { auth, domain } = this.props;
    console.log('App.js::Render::', domain);

      return (this.props.domainChecked ? 
        <MainErrorBoundary>
          <div className="App">
            <SecuredRoute
                path="/"
                component={PostAuthShell}
                authenticated={auth}
                domain={domain}
            />
          </div>
        </MainErrorBoundary> 
        : null
      );
  }
}

export const mapStateToProps = (state) => {
  return {
    auth: state.firebase.auth,
    domain: state.domain.domainId,
    domainChecked: state.domain.domainChecked
  }
}

export const mapDispatchToProps = (dispatch) => {
  return {
      SetDomain: () => dispatch(setDomain())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
