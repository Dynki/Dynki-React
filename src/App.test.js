import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps, mapStateToProps } from './App';

const mockOnSetDomainFn = jest.fn();

it('should render full component when domain has been checked', () => {
  const wrapper = shallow(<App SetDomain={mockOnSetDomainFn} domainChecked={true}/>);
  expect(wrapper).toMatchSnapshot();
});

it('should render null when domain not checked', () => {
  const wrapper = shallow(<App SetDomain={mockOnSetDomainFn} domainChecked={false}/>);
  expect(wrapper).toMatchSnapshot();
});

it('should set up state', () => {
  const initialState = {
    firebase: { auth: '1234' },
    domain: { domainId: 'ABC', domainChecked: true },
  }

  expect(mapStateToProps(initialState).auth).toEqual('1234');
  expect(mapStateToProps(initialState).domain).toEqual('ABC');
  expect(mapStateToProps(initialState).domainChecked).toEqual(true);
});

it('should set up dispatch', () => {
  const dispatch = jest.fn();

  mapDispatchToProps(dispatch).SetDomain();
  expect(dispatch).toBeCalled();
});