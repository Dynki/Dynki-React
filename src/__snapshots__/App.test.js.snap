// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render full component when domain has been checked 1`] = `
<MainErrorBoundary>
  <div
    className="App"
  >
    <SecuredRoute
      component={[Function]}
      path="/"
    />
  </div>
</MainErrorBoundary>
`;

exports[`should render null when domain not checked 1`] = `""`;
